src/material/schematics/ng-update/migrations/misc-checks/misc-class-inheritance.ts

```diff
@@ -6,14 +6,14 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {determineBaseTypes, Migration, TargetVersion} from '@angular/cdk/schematics';
+import {determineBaseTypes, DevkitMigration, TargetVersion} from '@angular/cdk/schematics';
 import * as ts from 'typescript';
 
 /**
  * Migration that checks for classes that extend Angular Material classes which
  * have changed their API.
  */
-export class MiscClassInheritanceMigration extends Migration<null> {
+export class MiscClassInheritanceMigration extends DevkitMigration<null> {
```

Comment:
Why have these been changed to a `DevkitMigration`? Is it due to the `never/unknown` change above for `Migration`, where they no longer are subtypes of `DevkitMigration` now? (this is what I wanted to achieve with `never` in the past).

The goal is to keep them decoupled from `DevkitMigration` so that these could technically be run within Google where they do not run within a devkit-based project.

Reply:
Yeah, pretty much. I noticed that the `DevkitMigration` doesn't do all that much on top of the `Migration` so I decided to swap them out. If you think that it's not the correct approach, I can try something else.

Reply:
It seems honestly unlikely at the current state that any of the migrations are needed in g3, but we put a lot of work initially into the infrastructure so that we can run them easily in g3 (without having to replicate logic w/ TSlint as in FW), so it would cool to see if we keep it like that, but if it's too much work then I personally don't mind a lot.

Reply:
I pushed another change to default the `Context` to `any` instead of `unknown`.

Reply:
👍 looks great!